### ISSUE TITLE
YouTube videos buffer after updating to 0.23.0

### ISSUE THREAD

**Issue by Taratect** [TeamNewPipe/NewPipe#8345]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.23.0

### Steps to reproduce the bug

Just watch any video on app. It buffers continuously ruining user experience.

Pro tip: To reproduce the bug easily, increase the time speed to 2x and above. I watch videos using 2.25x speed. Even though it buffers at 1x it's easier to notice the stutter at high timespeed.

### Expected behavior

The video should load seamlessly without any buffers.

### Actual behavior

The video buffers after 20 to 30 sec intervals. I have a stable internet and youtube app or website works fine. But Newpipe buffers. Normally, while watching YouTube my internet speed is bout 1 mbps+ however after the latest update it averages bout 250 kbps and peaking at 500 kpbs.

### Affected Android/Custom ROM version

Android 10/ MIUI 12

### Affected device model

Redmi 8

### Additional information

I have downgraded to 0.22.2 version. All buffering problems is solved. I think the bug is isolated to latest version (0.23.0). 
My speed also shows promising results of 1.5 mbps whereas it showed peaks of 500 kbps at latest version

**Comment by TiA4f8R** [comment]
Read and follow template instructions before opening an issue the next time, please.

> - [x] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).

Duplicate of #6949, which is pinned.

Thank you.

**Comment by opusforlife2** [comment]
@Taratect Could you try playing around with the Playback Interval setting? We didn't change anything related to buffering in 0.23.0.

**Comment by Taratect** [comment]
@opusforlife2 , I have updated the app again to 0.23.0 to use the playback interval load setting. I tried every option there (1-256 kb) but it still buffers for me. The internet speed showed around 300kb avg. I even tried to watch videos that played fully without buffering before updating in 0.22.2 but buffers in 0.23.0
I have used the 0.22.2 version for 6+ hours now but not a single vid buffered for once. The net speed was also good at 1 mbps+ while using app.
So, i think whatever the problem is, it lies on 0.23.0 version for me. The 0.22.2 version is working seamlessly!

**Comment by opusforlife2** [comment]
Alright. That was the extent of my knowledge. A dev will need to see this.

**Comment by litetex** [comment]
I think this might be a duplicate of #8238

**Comment by dev-a** [comment]
I'm also used to watch my videos in acceleration
I had no particular problems with version 0.22
On the other hand I have the same type of problem as described here with version 0.23 
I sometimes have to lower the quality to 144p,
I tried different things like changing the video format or changing the "playback load interval size" without seeing any noticeable change

**Comment by Chemus** [comment]
I have the exact same issue on Android 11/Galaxy A51

That is, on version 22.2, 3x playback functioned well at 720 (30fps), and on version 23.0, 1.5x-3x buffers after the initial buffer runs out.

Also tried changing the interval size to no effect.

**Comment by mateusviccari** [comment]
I can confirm this is happening only with 0.23, even with 140p videos it pauses frequently to buffer. On 0.22.2 I'm able to watch full hd videos as 3x speed without buffering issues. Using Android 11 (Lineage OS) on a Xiaomi Mi A2.

**Comment by TheArtForm** [comment]
Can confirm this bug for me as well. Pixel 6 on Android 12. Immediately after upgrading to .23 every video hit a buffer load screen after expending it's initial buffer. It does this constantly through the entire video every time. A downgrade to .22 fixed the problem immediately.

**Comment by TiA4f8R** [comment]
Video-only and separate audio streams are now preferred instead of video with audio streams. The second ones, legacy streams used as fallback and maybe for downloading by YouTube clients<sup>1</sup>, are less throttled (because of the ratebypass parameter) and are now only used if their is no video-only and separate audio streams available at a specific resolution.
This change has been done in order to allow the seamless transition, when leaving the app for instance.

But there is another change in 0.23.0, which in fact fixes an issue introduced in 0.22.2: usage of streams of the `ANDROID` client first (instead of a random client), which were known to be not throttled. It seems to be not the case anymore, at least when you don't follow the way the client is using to fetch streams.

ExoPlayer has been also updated to a newest version, who knows if it changed something related to media downloading/loading?

So, in order to improve the situation, you can:

- try to block the `youtubei.googleapis.com` domain for NewPipe, which will make only usage of streams of the `WEB` client and remove the ability to get the 720p stream with audio and video of musical contents and the 3GPP 144p stream, reduce available rewind time of livestreams (up to 30s) (the same domain is used to get a M3U8 link with a longer rewind time (from the iOS client)). If the decryption of the `n` parameter is broken, you will also get throttled streams;
- use the APK of #8153 (work still not final), which makes use of the same way as official clients to fetch streams and should improve loading times;
- rollback to version 0.22.2 (according to comments).

<sup>1</sup> one of them has issues on some videos right now, itag 22/720p (connection is closed by Google video servers after a small response). The issue seems to happen on recent videos after a short time the video was uploaded and fixes itself by waiting some days.

**Comment by nikhilCad** [comment]
@TiA4f8R the apk in the PR you have listed seems to work good, but because its a debug build the overall app is slow. Can you please provide a temporary release build?

**Comment by SC1040-TS2** [comment]
> Video-only and separate audio streams are now preferred instead of video with audio streams. The second ones, legacy streams used as fallback and maybe for downloading by YouTube clients1, are less throttled (because of the ratebypass parameter) and are now only used if their is no video-only and separate audio streams available at a specific resolution. This change has been done in order to allow the seamless transition, when leaving the app for instance.
> 
> But there is another change in 0.23.0, which in fact fixes an issue introduced in 0.22.2: usage of streams of the `ANDROID` client first (instead of a random client), which were known to be not throttled. It seems to be not the case anymore, at least when you don't follow the way the client is using to fetch streams.
> 
> ExoPlayer has been also updated to a newest version, who knows if it changed something related to media downloading/loading?
> 
> So, in order to improve the situation, you can:
> 
> * try to block the `youtubei.googleapis.com` domain for NewPipe, which will make only usage of streams of the `WEB` client and remove the ability to get the 720p stream with audio and video of musical contents and the 3GPP 144p stream, reduce available rewind time of livestreams (up to 30s) (the same domain is used to get a M3U8 link with a longer rewind time (from the iOS client)). If the decryption of the `n` parameter is broken, you will also get throttled streams;
> * use the APK of [Support delivery methods other than progressive HTTP #8153](https://github.com/TeamNewPipe/NewPipe/pull/8153) (work still not final), which makes use of the same way as official clients to fetch streams and should improve loading times;
> * rollback to version 0.22.2 (according to comments).
> 
> 1 one of them has issues on some videos right now, itag 22/720p (connection is closed by Google video servers after a small response). The issue seems to happen on recent videos after a short time the video was uploaded and fixes itself by waiting some days.

Question: Which version of ExoPlayer did NewPipe version 22.2 use?

**Comment by Chemus** [comment]
> Video-only and separate audio streams are now preferred instead of video with audio streams. The second ones, legacy streams used as fallback and maybe for downloading by YouTube clients1, are less throttled (because of the ratebypass parameter) and are now only used if their is no video-only and separate audio streams available at a specific resolution. This change has been done in order to allow the seamless transition, when leaving the app for instance.
> 
> But there is another change in 0.23.0, which in fact fixes an issue introduced in 0.22.2: usage of streams of the `ANDROID` client first (instead of a random client), which were known to be not throttled. It seems to be not the case anymore, at least when you don't follow the way the client is using to fetch streams.
> 
> ExoPlayer has been also updated to a newest version, who knows if it changed something related to media downloading/loading?
> 
> So, in order to improve the situation, you can:
> 
> * try to block the `youtubei.googleapis.com` domain for NewPipe, which will make only usage of streams of the `WEB` client and remove the ability to get the 720p stream with audio and video of musical contents and the 3GPP 144p stream, reduce available rewind time of livestreams (up to 30s) (the same domain is used to get a M3U8 link with a longer rewind time (from the iOS client)). If the decryption of the `n` parameter is broken, you will also get throttled streams;
> * use the APK of [Support delivery methods other than progressive HTTP #8153](https://github.com/TeamNewPipe/NewPipe/pull/8153) (work still not final), which makes use of the same way as official clients to fetch streams and should improve loading times;
> * rollback to version 0.22.2 (according to comments).
> 
> 1 one of them has issues on some videos right now, itag 22/720p (connection is closed by Google video servers after a small response). The issue seems to happen on recent videos after a short time the video was uploaded and fixes itself by waiting some days.

The debug apk appears to solve the issue.

**Comment by SC1040-TS2** [comment]
> > Question: Which version of ExoPlayer did NewPipe version 22.2 use?
> 
> 0.22.2 uses version 2.14.2 of ExoPlayer and 0.23.0 is using version 2.17.1.
> 
> Also, please don't quote full comments when they are big, but only the relevant part next time (sentence applies for both SC1040-TS2 and Chemus). Thank you in advance!

Bad habit from other sites. Oh well.

Anyway, in briefly skimming through the ExoPlayer release notes found here: https://github.com/google/ExoPlayer/releases , it appears that a lot of function renaming, component moving and refactoring, and deprecation was done between ExoPlayer versions 2.14.2 and 2.17.1. This may not have been entirely accounted for in NewPipe's and the NewPipe Extractor's code.

There is also the leftover workaround for OkHttp's version upgrade and Android minSDK version upgrade in Exoplayer 2.15.0, which was implemented so NewPipe could continue to maintain support for older Android devices, in spite of ExoPlayer and F-Droid deprecating them all by now. Maybe the time has come to no longer beat around the bush there, seeing as legacy device users whom have Android 5.0 (minSDK 21) and under cannot install newer F-Droid versions since F-Droid 1.13 at all.

**Comment by SC1040-TS2** [comment]
Otherwise, the last section of the above reply is an unwarranted tangent.

**Comment by shadowcyberdeck** [comment]
Can also confirm this issue with v0.23 of NP, on Galaxy Note 9 android 10 OneUI 2.5. Nothing seems to resolve the problem, will try the debug apk in the post above, and report back.

**Comment by lordmvk** [comment]
Can confirm same issue with Motorola g40, downgraded to 0.22.2  version.

**Comment by TiA4f8R** [comment]
> Can't find the debug APK, ended up rolling back to the previous version for now

Go the `Checks` tab of the pull request and click on `CI on pull_request`, then scroll down to the `Artifacts` section. Note that if you are not logged in to GitHub, you will not find it.

**Comment by shadowcyberdeck** [comment]
> > Can't find the debug APK, ended up rolling back to the previous version for now
> 
> Go the `Checks` tab of the pull request and click on `CI on pull_request`, then scroll down to the `Artifacts` section. Note that if you are not logged in to GitHub, you will not find it.

Found it, thanks. Testing so far seems to work fine at initial testing.

**Comment by AudricV** [comment]
Please try the release candidate, available in #8548. Does the issue is improved?

**Comment by CBSkarmory** [comment]
yes this appears to fix the issue

**Comment by tsiflimagas** [comment]
So, is this considered to be fixed in 0.23.1?

**Comment by mateusviccari** [comment]
> Please try the release candidate, available in #8548. Does the issue is improved?

Ive been using the RC for the past two weeks and everything works perfectly, I'm yet to see any buffering message.

**Comment by opusforlife2** [comment]
Alright. Closing this. Please open a new issue if needed.

**Comment by jcperil** [comment]
Hey, I am facing this slow loading issue in v0.24.1. It had to do with `googleapis.com`.

I disabled `googleapis.com` for NewPipe via TrackerControl. The video took 79 secs to load.

https://user-images.githubusercontent.com/89805719/213683406-6c5bf0b6-4fdc-4b89-8986-c0c13531b050.mp4

When I enabled `googleapis.com` for NewPipe, the same video took 7 secs to load.

https://user-images.githubusercontent.com/89805719/213681882-19303ccd-2eb1-4e73-9643-75e3eb8103d5.mp4
